webui/portal/src/main/java/org/exoplatform/portal/application/localization/DefaultLocalePolicyService.java

```diff
@@ -47,6 +64,48 @@
  * @author <a href="mailto:mstrukel@redhat.com">Marko Strukelj</a>
  */
 public class DefaultLocalePolicyService implements LocalePolicy, Startable {
```

Comment:
Not sure how it relates to my comment. I am talking about creating a new service that will collect all the required information to calculate the locale (build a LocaleContextInfo object), then update (if necessary) the DefaultLocalePolicyService service to update the algorithm which calcultes the locale.
Here you add (static !) methods directly in the implementation (so without adding the methods in the interface) which makes it not pluggable anymore. And therefore, you have to hardcode its use : Locale locale = DefaultLocalePolicyService.getLocaleFromState(state). With this design I can no longer customize the default locale policy in my custom extension.
Do not hesitate to ping me if you want to talk about that in details.